### PR TITLE
Added Elasticsearch expression support

### PR DESCRIPTION
**Pull request by sohansevta**
**1. Issue Link:** 
#32 

**2. Brief explanation of a change:** 
Added support to create Elasticsearch criteria instance to be able to use graphQL generic filters with MongoDB.

**3. Will it break existing clients and code in production?**
No

### CODE REVIEW

**Comment by sohanlal311** [comment]
@jaiswalsk Could you please review this PR?

File: src/main/java/com/intuit/graphql/filter/visitors/ElasticsearchCriteriaExpressionVisitor.java [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package com.intuit.graphql.filter.visitors;
+
+import com.intuit.graphql.filter.ast.*;
```
Comment (jaiswalsk) [inline src/main/java/com/intuit/graphql/filter/visitors/ElasticsearchCriteriaExpressionVisitor.java:18]: we can replace ```*``` with specific required imports.

File: src/main/java/com/intuit/graphql/filter/visitors/ElasticsearchCriteriaExpressionVisitor.java [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+            case ENDS:
+                criteria = Criteria.where(fieldName).endsWith(operandValue.value().toString());
+                break;
+
+            case CONTAINS:
+                criteria = Criteria.where(fieldName).contains(operandValue.value().toString());
+                break;
+
+            case EQUALS:
+                //criteria = Criteria.where(fieldName).is(operandValue.value());
```
Comment (jaiswalsk) [inline src/main/java/com/intuit/graphql/filter/visitors/ElasticsearchCriteriaExpressionVisitor.java:119]: pls remove this commented line.

**Review by jaiswalsk — CHANGES_REQUESTED** [review]
Thank you @sohanlal311 for your contributions! the PR looks good, except couple of minor changes that I have requested.

File: src/main/java/com/intuit/graphql/filter/visitors/ElasticsearchCriteriaExpressionVisitor.java (hunk 1 above)
Comment (sohanlal311) [reply src/main/java/com/intuit/graphql/filter/visitors/ElasticsearchCriteriaExpressionVisitor.java]: Its done.

File: src/main/java/com/intuit/graphql/filter/visitors/ElasticsearchCriteriaExpressionVisitor.java (hunk 2 above)
Comment (sohanlal311) [reply src/main/java/com/intuit/graphql/filter/visitors/ElasticsearchCriteriaExpressionVisitor.java]: Its done.

**Comment by sohanlal311** [comment]
@jaiswalsk Thanks for your review. I have done the changes and also squashed the commits into one. Please review.

**Pull request merged**